aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -102,6 +107,66 @@ void convert_indices_from_csr_to_coo_cuda(const Tensor& indices, const Tensor& c
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+template <typename scalar_t>
+void _apply_sparse_csr_qr_solve(
+  const Tensor& input,
+  const Tensor& other,
+  const Tensor& result,
+  int &_singularity) {
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+  auto values = input.values();
+  const scalar_t *values_data_ptr = values.data_ptr<scalar_t>();
+  auto crow_indices = input.crow_indices().to(kInt);
+  const int *crow_indices_data_ptr = crow_indices.data_ptr<int>();
+  auto col_indices = input.col_indices().to(kInt);
+  const int *col_indices_data_ptr = col_indices.data_ptr<int>();
+  auto handle = at::cuda::getCurrentCUDASolverSpHandle();
+  auto descrA = at::cuda::sparse::CuSparseMatDescriptor();
+
+  const scalar_t *b = other.data_ptr<scalar_t>();
+  int n = crow_indices.numel() - 1;
```

Comment:
We can use `.size` instead. And let's use `cuda_int_cast()` for all integers that are passed to cusolver (n, nnz are stored as `int64_t` type and we cast it into `int` for larger sizes it might lead to silent problems).
```suggestion
  int n = cuda_int_cast(input.size(-1), "n");
```

Reply:
Oh, interesting. I didn't know about `cuda_int_cast`, thanks Ivan. Fixed now.